aws/get_running_instances.py

```diff
@@ -0,0 +1,88 @@
+"""
+Use boto to get the number of running or pending EC2 instances matching a key
+name.
+
+Usage:
+    If you've defined the AWS credentials as environment variables or in a
+    .boto file, then use:
+        `python get_running_instances.py -k some-key-name`
+
+    Else, you can add the aws keys as arguments to the above command:
+        ** '--aws-access-key-id' or '-i'
+        ** '--aws-secret-access-key' or '-s'
+"""
+import argparse
+import boto
+import logging
+import os
+import sys
+
+
+logging.basicConfig(format='[%(levelname)s] %(message)s')
+logger = logging.getLogger(__name__)
+
+
+def get_running_instances(key_name, aws_access_key_id, aws_secret_access_key):
+    """
+    Logs and returns (as a string) the number of EC2 instances matching
+    `key_name` that are running or pending.
+
+    :Args:
+        key_name: the ec2 key_name to match.
```

Comment:
key_name is a bit tricky, because someone could (intentionally or inadvertently) use that keypair when spinning up a new ami manually to test something out. Though I really don't know what to suggest instead, as the same would hold true for IAM role, etc. Maybe this is just something we need to be aware of. @benpatterson what do you think?